### PR TITLE
[AEA-712] Framework core better test coverage

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

improved test coverage for framework core

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/aea_builder.py [hunk 1]
```diff
@@ -237,7 +238,9 @@ def pypi_dependencies(self) -> Dependencies:
         return all_pypi_dependencies
 
     @staticmethod
-    def _build_dotted_part(component, relative_import_path) -> str:
+    def _build_dotted_part(
+        component, relative_import_path: str
+    ) -> str:  # pragma: nocover # TODO: remove? not in use at the moment!
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:243]: @marcofavorito can it be removed?

File: aea/aea_builder.py [hunk 2]
```diff
@@ -515,6 +524,7 @@ def _check_can_remove(self, component_id: ComponentId) -> None:
         :return: None
         :raises ValueError: if the component is already present.
         """
+        print(self._package_dependency_manager.all_dependencies)
```
Comment (DavidMinarsch) [inline aea/aea_builder.py:527]: Please remove print statement or make log debug statement

File: aea/agent_loop.py [hunk 3]
```diff
@@ -263,10 +263,6 @@ async def _task_process_inbox(self) -> None:
         self.logger.info("[{}]: Start processing messages...".format(self._agent.name))
         while self.is_running:
             await inbox.async_wait()
-
-            if not self.is_running:  # make it close faster
-                return
-
```
Comment (DavidMinarsch) [inline aea/agent_loop.py:269]: No longer needed?

File: aea/runtime.py [hunk 4]
```diff
@@ -152,9 +154,6 @@ def _start(self) -> None:
 
         Start runtime asynchonously in own event loop.
         """
-        if self._state.get() is RuntimeStates.started:
-            raise ValueError("Runtime already started!")
```
Comment (DavidMinarsch) [inline aea/runtime.py:156]: why removed?

File: aea/aea_builder.py (hunk 1 above)
Comment (marcofavorito) [reply aea/aea_builder.py:244]: If it is not used elsewhere then yes

File: aea/agent_loop.py (hunk 3 above)
Comment (solarw) [reply aea/agent_loop.py:269]: it's was additional i check, just decided to remove

File: aea/runtime.py (hunk 4 above)
Comment (solarw) [reply aea/runtime.py:156]: cause the same check is performed in caller method. double check.

File: aea/aea_builder.py (hunk 1 above)
Comment (solarw) [reply aea/aea_builder.py]: removed

File: aea/aea_builder.py (hunk 2 above)
Comment (solarw) [reply aea/aea_builder.py]: done

**Pull request merged**